x-pack/test/functional/apps/canvas/custom_elements.ts

```diff
@@ -0,0 +1,164 @@
+      await PageObjects.canvas.openAddElementModal();
+
+      // open the custom elements tab
+      const elem = await find.byCssSelector('#customElements', 20000);
+      await elem.click();
+
+      // ensure the custom element is the one expected and click it to add to the workpad
+      const customElement = await find.byCssSelector(
+        '[aria-labelledby="customElements"] .canvasElementCard__wrapper'
+      );
+      const elementName = await customElement.findByCssSelector('.euiCard__title');
+      expect(await elementName.getVisibleText()).to.contain('My New Element');
+      customElement.click();
+
+      await retry.try(async () => {
+        // ensure the new element is on the workpad
+        const elements = await testSubjects.findAll(
+          'canvasWorkpadPage > canvasWorkpadPageElementContent'
+        );
+        expect(elements).to.have.length(5);
+
+        // ensure the new element has the expected content
+        const newElem = await elements[4];
+        const md = await newElem.findByCssSelector('.canvasMarkdown');
+        expect(await md.getVisibleText()).to.contain('Welcome to Canvas');
+
+        // delete the element off the workpad
+        await newElem.click();
+        await browser.pressKeys(browser.keys.DELETE);
+
+        // ensure the new element has been removed
+        const elementsAgain = await testSubjects.findAll(
+          'canvasWorkpadPage > canvasWorkpadPageElementContent'
+        );
+        expect(elementsAgain).to.have.length(4);
+      });
+    });
+
+    it('saves custom element modifications', async () => {
+      await PageObjects.canvas.openAddElementModal();
+
+      // open the custom elements tab
+      const elem = await find.byCssSelector('#customElements', 20000);
+      await elem.click();
+
+      // ensure the correct amount of custom elements exist
+      const container = await find.byCssSelector('[aria-labelledby="customElements"]');
+      const customElements = await container.findAllByCssSelector('.canvasElementCard__wrapper');
+      expect(customElements).to.have.length(1);
+
+      // hover over the custom element to bring up the edit and delete icons
+      const customElement = customElements[0];
+      await customElement.moveMouseTo();
+
+      // click the edit element button
+      const editBtn = await customElement.findByCssSelector('[aria-label="Edit element"]');
+      await editBtn.click();
+
+      // fill out the custom element form and submit it
+      await PageObjects.canvas.fillOutCustomElementForm(
+        'My Edited New Element',
+        'An excellent edited element'
+      );
+
+      // ensure the custom element in the modal shows the updated text
+      await retry.try(async () => {
+        const customElementAgain = await find.byCssSelector(
+          '[aria-labelledby="customElements"] .canvasElementCard__wrapper'
+        );
+
+        const elementName = await customElementAgain.findByCssSelector('.euiCard__title');
```

Comment:
it is better to search for child element directly in 1 call to make things faster and more stable:
```
 const elementName = await find.byCssSelector('[aria-labelledby="customElements"] .canvasElementCard__wrapper .euiCard__title');
```